### PR TITLE
Adding some pedagogy to Firefly notebook

### PR DESCRIPTION
**Pull request by kadrlica**
This is a running PR so that @SimonKrughoff and @drphilmarshall can see and comment on what I'm up to...

It is not ready to merge.

### CODE REVIEW

**Comment by kadrlica** [comment]
Ok @SimonKrughoff, I think this is ready for a review.

**Comment by kadrlica** [comment]
1) Is preferential ordering of sources from the bottom of the image upwards representative of the order in which source finding is done? I.e., is the order of the sources in the `src` table representative of some algorithmic sequence, or is there some re-ordering done?
2) What are the units of the `size` argument to `Display.dot`?

File: Firefly.ipynb [hunk 1]
```diff
@@ -62,22 +93,72 @@
...
+   "source": [
+    "dataId = {'filter': 'r', 'raft': '2,2', 'sensor': '1,1', 'visit': 235}\n",
+    "calexp = butler.get('calexp', **dataId)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `calexp` is an `ExposureF` Python object. Exposures are powerful representations of image data because they contain not only the image data, but also a variance image for uncertainty propagation, a bit mask image plane, and key-value metadata. In the next steps you’ll learn how to display an Exposure’s image and mask."
```
Comment (SimonKrughoff) [inline Firefly.ipynb]: I'm not sure if you want to mention it here, but `ExposureF`s can also contain other things like PSF models and WCSs.

File: Firefly.ipynb [hunk 2]
```diff
@@ -160,22 +279,152 @@
...
+    "with afw_display.Buffering():\n",
+    "    for record in src[:100]:\n",
+    "        afw_display.dot('o', record.getX(), record.getY(), size=20, ctype='orange')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that sources that the first 100 sources are preferentially located at the bottom of the image. This spatial ordering is likely imprinted by the source detection algorithm; however, it could change due to parallelization. The units o the `size` parameter are believed to be pixels."
```
Comment (SimonKrughoff) [inline Firefly.ipynb]: Note that sources that the first --> Note that the first

File: Firefly.ipynb [hunk 3]
```diff
@@ -287,6 +545,29 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Wrap up\n",
+    "In this tutorial we have used the `Butler` to access LSST simulation data and used the LSST Science Pipelines Python API to display images and tables. Here are some key takeaways:\n",
+    "\n",
+    "* Use the `lsst.daf.persistence.Butler` class to read and write data from repositories.\n",
+    "* The `lsst.afw.display` module provides a flexible framework for sending data from LSST Science Pipelines code to image displays. We used the Firefly backend for web-based visualization (`ds9`, `matplotlib`, and **???** are other avialable backends).\n",
```
Comment (SimonKrughoff) [inline Firefly.ipynb]: I believe `ginga` is also available.

**Comment by kadrlica** [comment]
Thanks @SimonKrughoff. Changes made.

**Comment by SimonKrughoff** [comment]
Thanks again for doing this.  It's actually our first upstreamed change on a notebook and I think it shows the process works pretty well.

**Pull request merged**